proximo-server/kafka.go

```diff
@@ -143,3 +145,34 @@ func (h *kafkaHandler) HandleProduce(ctx context.Context, topic string, forClien
 		}
 	}
 }
+
+func (h *kafkaHandler) Status() (bool, []error) {
+	errs := make(chan error)
+
+	for _, broker := range h.brokers {
+		go func(broker string) {
+			conn, err := net.DialTimeout("tcp", broker, 10*time.Second)
+			if err != nil {
+				errs <- fmt.Errorf("Failed to connect to broker %s: %v", broker, err)
+				return
+			}
+			if err = conn.Close(); err != nil {
+				errs <- fmt.Errorf("Failed to close connection to broker %s: %v", broker, err)
+				return
+			}
+			errs <- nil
+		}(broker)
+	}
+
+	errors := []error{}
+	for range h.brokers {
+		err := <-errs
```

Comment:
It's probably worth doing a `select` in here so we can time out if we don't get a response in a reasonable time

Reply:
@ggordan line 154 implements timeout in connection.